solidity/test/rewards/TestECDSARewards.js

```diff
@@ -0,0 +1,232 @@
+const {accounts, contract, web3} = require("@openzeppelin/test-environment")
+const {createSnapshot, restoreSnapshot} = require("../helpers/snapshot")
+const {initialize, fund, createMembers} = require("./rewardsSetup")
+
+const {expectRevert, time} = require("@openzeppelin/test-helpers")
+
+const BondedECDSAKeepStub = contract.fromArtifact("BondedECDSAKeepStub")
+const ECDSARewards = contract.fromArtifact("ECDSARewards")
+
+const BN = web3.utils.BN
+
+const chai = require("chai")
+chai.use(require("bn-chai")(BN))
+const expect = chai.expect
+
+describe("ECDSARewards", () => {
+  let rewardsContract
+  let keepToken
+
+  let tokenStaking
+  let keepFactory
+  let keepMembers
+  let expectedKEEPAllocation
+
+  const owner = accounts[0]
+
+  // Solidity is not very good when it comes to floating point precision,
+  // we are allowing for ~1 KEEP difference margin between expected and
+  // actual value.
+  const precision = 1
+  const tokenDecimalMultiplier = web3.utils.toBN(10).pow(web3.utils.toBN(18))
+  const firstKeepCreationTimestamp = 1600041600 // Sep 14 2020
+
+  // 1,000,000,000 - total KEEP supply
+  //   200,000,000 - 20% of the total supply goes to staker rewards
+  //   180,000,000 - 90% of staker rewards goes to the ECDSA staker rewards
+  //   178,200,000 - 99% of ECDSA staker rewards goes to keeps opened after Sep 14 2020
+  const KEEPRewards = web3.utils.toBN(178200000).mul(tokenDecimalMultiplier)
+
+  before(async () => {
+    const setup = await initialize()
```

Comment:
Maybe `const contracts`? This function returns three contracts.

Reply:
Ok.